src/Editor/Editor.cpp

```diff
@@ -245,20 +245,30 @@ void Editor::Show(float deltaTime) {
         if (Input()->Pressed(InputHandler::SELECT) && !ImGui::IsMouseHoveringAnyWindow()) {
             mousePicker.UpdateProjectionMatrix(cameraEntity->GetComponent < Component::Lens>()->GetProjection(glm::vec2(MainWindow::GetInstance()->GetSize().x, MainWindow::GetInstance()->GetSize().y)));
             mousePicker.Update();
-
-            for (int i = 0; i < Hymn().world.GetEntities().size(); ++i) {
+            float lastDistance = INFINITY;
+            int entityIndex = 0;
+            int entityAmount = Hymn().world.GetEntities().size();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The scope of the variable 'intersectDistance' can be reduced. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AvariableScope)